impress-org/givewp
# DI Container auto wiring now works correctly when using PHP8

Resolves #5987

## Description

On some rare occasions, [$dependency->getClass() ](https://github.com/impress-org/givewp/blob/123d168ce0966409e96dcf0ad6cb2dc7da215085/src/Container/Container.php#L745) and [$dependency->getType()](https://github.com/impress-org/givewp/blob/123d168ce0966409e96dcf0ad6cb2dc7da215085/src/Container/Container.php#L746) can return `null`. In my understanding, this happens when the dependency class has optional parameters in constructor, like [ApiKey](https://github.com/impress-org/give-mailchimp/blob/bdbaed4a760addf95f1ef1ac91dba8152b136e5b/src/MailChimp/Repositories/ApiKey.php#L23) class has from MailChimp add-on. 

This PR resolves the issue by adding additional `is_null` checks to `Give\Conatiner::resolveDependencies()` and `Give\Container::resolveClass()` methods.

## Affects

Give\Container

## Testing Instructions

1. Make sure your site is running using PHP8
2. Download and install the MailChimp add-on
3. Activate it




## Pre-review Checklist
